File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.py
def solution(n):
    result = ""
    while n > 0:
        # 3으로 나눈 나머지를 문자열 앞에 붙임
        result = str(n % 3) + result
        # 3으로 나눈 몫으로 갱신
        n //= 3
    result = result[::-1]

    return int(result, 3)

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        StringBuilder sb = new StringBuilder();
        
        while(n > 0){
            sb.append(n % 3);
            n /= 3;
        }
        for(int i=0; i<sb.length(); i++){
            int num = sb.charAt(i) - '0';
            answer = answer * 3 + num;
        }
        return answer;
    }
}